gateway/src/apicast/policy/routing/apicast-policy.json

```diff
@@ -0,0 +1,154 @@
+{
+  "$schema": "http://apicast.io/policy-v1/schema#manifest#",
+  "name": "Routing",
+  "summary": "Allows to modify the upstream URL of the request.",
+  "description": [
+    "This policy allows to modify the upstream URL (scheme, host and port) of ",
+    "the request based on its path, its query arguments, a header, or a JWT ",
+    "claim. \n",
+    "When combined with the APIcast policy, the upstream policy should be ",
+    "placed before it in the policy chain."
+  ],
+  "version": "builtin",
+  "configuration": {
+    "type": "object",
+    "definitions": {
+      "operation": {
+        "type": "object",
+        "$id": "#/definitions/operation",
+        "properties": {
+          "thing_to_match": {
+            "type": "string",
+            "enum": [
+              "path",
+              "header",
+              "query_arg",
+              "jwt_claim"
+            ]
+          },
+          "op": {
+            "type": "string",
+            "enum": [
+              "==",
+              "!=",
+              "matches"
+            ]
+          },
+          "value": {
+            "type": "string"
+          },
+          "value_type": {
+            "description": "How to evaluate 'type'",
+            "type": "string",
+            "default": "plain",
+            "oneOf": [
+              {
+                "enum": ["plain"],
+                "title": "Evaluate 'value' as plain text."
+              },
+              {
+                "enum": ["liquid"],
+                "title": "Evaluate 'value' as liquid."
+              }
+            ]
+          }
+        },
+        "required": [
+          "thing_to_match",
+          "op",
+          "value"
+        ],
+        "dependencies": {
+          "thing_to_match": {
```

Comment:
That is a bit weird property name, no?

Reply:
I'd suggest `property`, `match`, `key_name` (that would match 1:1 to the key name). Not really sure, naming things is hard :)

Reply:
`match` looks OK to me. I'll change it.

Reply:
Done: https://github.com/3scale/apicast/pull/976/commits/71e16d98c5378bc3b6e99e8930264541e8799309